cub/test/catch2_test_device_merge_sort.cu

```diff
@@ -0,0 +1,309 @@
+  c2h::gen(CUB_SEED(2), key_ranks, offset_t{}, static_cast<offset_t>(128));
+  thrust::device_vector<thrust::tuple<key_t, offset_t>> keys_in(num_items);
+  auto sort_key_it = thrust::make_transform_iterator(key_ranks.begin(), rank_to_key_op_t<offset_t, key_t>{});
+  auto keys_in_it  = thrust::make_zip_iterator(sort_key_it, key_ranks.begin());
+  thrust::copy(keys_in_it, keys_in_it + num_items, keys_in.begin());
+
+  // Perform sort
+  thrust::device_vector<thrust::tuple<key_t, offset_t>> keys_out(num_items);
+  thrust::fill(keys_out.begin(),
+               keys_out.end(),
+               thrust::tuple<key_t, offset_t>{rank_to_key_op_t<offset_t, key_t>{}(42), static_cast<offset_t>(42)});
+  stable_sort_keys_copy(
+    thrust::raw_pointer_cast(keys_in.data()),
+    thrust::raw_pointer_cast(keys_out.data()),
+    num_items,
+    compare_first_lt_op_t{});
+
+  // Verify results
+  thrust::host_vector<thrust::tuple<key_t, offset_t>> keys_expected(keys_in);
+  std::stable_sort(keys_expected.begin(), keys_expected.end(), compare_first_lt_op_t{});
+
+  REQUIRE(keys_expected == keys_out);
+}
+
+CUB_TEST("DeviceMergeSort::StableSortKeys works", "[merge][sort][device]", key_types)
+{
+  using key_t    = typename c2h::get<0, TestType>;
+  using offset_t = std::int32_t;
+
+  // Prepare input
+  const offset_t num_items = GENERATE_COPY(take(2, random(1, 1000000)), values({500, 1000000, 2000000}));
+  thrust::device_vector<key_t> sort_keys(num_items);
+  c2h::gen(CUB_SEED(2), sort_keys);
+  auto keys_in_it = thrust::make_zip_iterator(sort_keys.begin(), thrust::make_counting_iterator(offset_t{}));
+  thrust::device_vector<thrust::tuple<key_t, offset_t>> keys_in_out(num_items);
+  thrust::copy(keys_in_it, keys_in_it + num_items, keys_in_out.begin());
+
+  // Perform sort
+  stable_sort_keys(thrust::raw_pointer_cast(keys_in_out.data()), num_items, compare_first_lt_op_t{});
+
+  // Verify results
+  thrust::host_vector<thrust::tuple<key_t, offset_t>> keys_expected(keys_in_out);
+  std::stable_sort(keys_expected.begin(), keys_expected.end(), compare_first_lt_op_t{});
+
+  REQUIRE(keys_expected == keys_in_out);
+}
+
+CUB_TEST("DeviceMergeSort::SortPairsCopy works", "[merge][sort][device]", wide_key_types)
+{
+  using key_t    = typename c2h::get<0, TestType>;
+  using offset_t = std::int32_t;
+
+  // Prepare input
+  const offset_t num_items = GENERATE_COPY(take(2, random(1, 1000000)), values({500, 1000000, 2000000}));
+  auto key_ranks           = make_shuffled_key_ranks_vector(num_items, CUB_SEED(2));
+  thrust::device_vector<key_t> keys_in(num_items);
+  thrust::transform(key_ranks.begin(), key_ranks.end(), keys_in.begin(), rank_to_key_op_t<offset_t, key_t>{});
+
+  // Perform sort
+  thrust::device_vector<key_t> keys_out(num_items);
+  thrust::device_vector<offset_t> values_out(num_items);
+  thrust::fill(keys_out.begin(), keys_out.end(), static_cast<key_t>(42));
+  thrust::fill(values_out.begin(), values_out.end(), static_cast<offset_t>(42));
```

Comment:
suggestion:
```suggestion
  thrust::device_vector<key_t> keys_out(num_items, static_cast<key_t>(42));
  thrust::device_vector<offset_t> values_out(num_items, static_cast<offset_t>(42));
```

Reply:
Thanks, following your suggestion in other places too. Value-initializing vectors instead of an extra `thrust::fill`.